src/analysis/transcendental/liouville_theorem.lean

```diff
@@ -0,0 +1,441 @@
+              conv_lhs { rw [eval_map, as_sum' f, eval₂_finset_sum] },
+              simp only [coeff_map, eval₂_X_pow, div_pow, eval₂_mul, eval₂_C],
+              congr
+            end
+      ... = 1/b^f.nat_degree * (∑ i in f.support, f.coeff i*a^i*b^(f.nat_degree - i)) :
+            begin
+              rw mul_sum, apply sum_congr rfl,
+              intros n h,
+              rw [calc (f.coeff n : ℝ) * (a/b)^n
+                      = f.coeff n * (a ^ n / b ^ n) : by rw div_pow
+                  ... = f.coeff n * a ^ n * (b ^ n)⁻¹ : by ring
+                  ... = f.coeff n * a ^ n * b ^ (- n : ℤ) : by simp only [fpow_neg, fpow_coe_nat],
+                  calc 1/(b : ℝ) ^ f.nat_degree * (f.coeff n * a ^ n * b ^ (f.nat_degree - n))
+                      = f.coeff n * a ^ n * (b^(f.nat_degree - n) / (b ^ f.nat_degree)) : by ring
+                  ... = f.coeff n * a ^ n * (b ^ ((f.nat_degree - n : ℕ) - f.nat_degree : ℤ))
+                      : by {rw fpow_sub, simp only [fpow_coe_nat], norm_cast, exact b_nonzero}
+                  ],
+              congr,
+              rw [neg_eq_iff_neg_eq, neg_sub, sub_eq_iff_eq_add],
+              norm_cast,
+              rw calc n + (f.nat_degree - n) = n + f.nat_degree - n
+                  : begin
+                      rw ←nat.add_sub_assoc, by_contra a,
+                      simp only [gt_iff_lt, not_le, finsupp.mem_support_iff, ne.def] at a,
+                      rw mem_support_iff_coeff_ne_zero at h,
+                      refine h _, rw coeff_eq_zero_of_nat_degree_lt a
+                    end
+                ...  = f.nat_degree + n - n : by rw add_comm
+                ...  = f.nat_degree + (n - n) : by rw nat.add_sub_assoc (rfl.ge)
+                ...  = f.nat_degree + 0 : by rw nat.sub_self
+                ...  = f.nat_degree : by rw add_zero,
+            end,
+have cast_eq : ∑ i in f.support, (f.coeff i : ℝ) * a ^ i * b ^ (f.nat_degree - i)
+            = ↑(∑ i in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)), from
+begin
+  induction f.support using finset.induction_on with a s H ih,
+  { simp only [int.cast_zero, sum_empty] },
+  { rw [sum_insert H, sum_insert H, ih],
+    simp only [int.cast_pow, int.cast_add, int.cast_mul] }
+end,
+begin
+  rw [eq₁, cast_eq, abs_mul, abs_of_pos pos, le_mul_iff_one_le_right pos],
+  norm_cast,
+  suffices : 0 < abs (∑ (i : ℕ) in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i)),
+  { omega },
+  { rw abs_pos_iff,
+    rw [eq₁, mul_ne_zero_iff, cast_eq] at not_root,
+    norm_cast at not_root,
+    exact not_root.2 }
+end
+
+lemma about_irrational_root (α : ℝ) (hα : irrational α) (f : polynomial ℤ)
+  (f_deg : 1 < f.nat_degree) (α_root : (f.map (algebra_map ℤ ℝ)).eval α = 0) :
+  ∃ A : ℝ, 0 < A ∧ ∀ a b : ℤ, 0 < b → (A / b ^ (f.nat_degree)) < abs(α - a / b)  :=
+have f_nonzero : f ≠ 0, from
+begin
+  intro rid,
+  have : f.nat_degree = 0 := (congr_arg polynomial.nat_degree rid).trans rfl,
+  rw this at f_deg, linarith,
+end,
```

Comment:
surely this should be a lemma somewhere? If not this could probably go in` data.polynomial.degree` as `ne_zero_iff_one_lt_nat_degree` (maybe the other way round)

Reply:
There is one about degree instead of nat_degree
```lean
lemma ne_zero_of_degree_gt {n : with_bot ℕ} (h : n < degree p) : p ≠ 0
```

Reply:
done